src/c360_client/dataset_cls/__init__.py

```diff
@@ -0,0 +1,233 @@
+import os
+import json
+import requests
+import getpass
+
+from c360_client.utils import get_boto_client
+
+
+class DatalakeClientDataset:
+    def __init__(
+        self, tenant=None, stage="prod", api_url=None, boto_endpoint_url=None, api_key=None
+    ):
+        """
+        """
+        self.api_key = api_key
+        self.tenant = tenant
+        self.stage = stage
+        self.url = api_url
+
+        self.boto_kwargs = {}
+        if boto_endpoint_url:
+            self.boto_kwargs["endpoint_url"] = boto_endpoint_url
+
+        # options
+        self._is_user_scoped = True
+
+        self._cached_user_scope = None
+
+    def set_options(self, is_user_scoped=True):
+        self._is_user_scoped = is_user_scoped
+
+    def set_api_key(self):
+        self.api_key = getpass.getpass("API_KEY:")
+
+    def get_groups(self, groups):
+        main_groups = []
+        if self._is_user_scoped:
+            main_groups.append("users")
+            main_groups.append(self._get_user_scope())
+
+        print(main_groups + groups)
+
+        return main_groups + groups
+
+    def _request(self, endpoint, **kwargs):
+
+        # use API key
+        if not kwargs.get("headers"):
+            kwargs["headers"] = {}
+
+        if not kwargs["headers"].get("Authorization"):
+            kwargs["headers"]["Authorization"] = self.api_key
+
+        req = requests.request(url=f"{self.url}/{endpoint}", **kwargs)
+        return req
```

Comment:
what if user doesn't provide valid api_key.

btw not for now but we may need to consider blocking access/limiting number of API calls per minute per user/IP address (ddos thing..)

Reply:
> what if user doesn't provide valid api_key.

API will return 401 unauthenticated

> btw not for now but we may need to consider blocking access/limiting number of API calls per minute per user/IP address (ddos thing..)

valid concern, although more fitting on the API repo than in this one